### DISCUSSION TITLE
not able to apply settings

### DISCUSSION THREAD

**Discussion by farhan-beg** [realmazharhussain/gdm-settings#176] [Q&A]
using it on kali linux when i try to apply settings it says failed to apply settings

**Comment by farhan-beg** [comment]
/root/.cache/app/io.github.realmazharhussain.GdmSettings/run-elevated: 1: cp: not found
this error shows up

**Reply by realmazharhussain** [reply]
Do not run this app as root. It will ask for your password and elevate its privileges when it needs to.

**Reply by farhan-beg** [reply]
I am not running it as root but as soon as i click apply settings it asks password then throws that error

**Reply by realmazharhussain** [reply]
The path in the error starts with `/root/.cache`. If run as a regular user, the path to that file usually starts with `/run/user/1000/` instead. So, there is a possibility that either gdm-settings is running as root without your knowledge or your system is messed up in some other way. In any case, I need more information to be able to know what is actually happening.

How did you install gdm-settings? Was it a deb package, AppImage, or Flatpak?

What does gdm-settings itself output if you run it with `-v` option?

What is the output of the following commands?

```bash
whoami
echo $XDG_RUNTIME_DIR
echo $XDG_CACHE_HOME
```

**Reply by farhan-beg** [reply]
app was installed in deb as well appimage this one is appimage i removed .deb bcoz of this nd tried appimage have another one with name login manager settings installed from software app having same issue that one is flatpack.
output of -v:
```
INFO:gdms.gui: Application Version    = 4.alpha
INFO:gdms.gui: Operating System       = Kali GNU/Linux Rolling
INFO:gdms.gui: PackageType            = Unknown
INFO:gdms.gui: TEMP_DIR               = /run/user/1000/app/io.github.realmazharhussain.GdmSettings
INFO:gdms.gui: HOST_DATA_DIRS         = /usr/local/share:/usr/share
INFO:gdms.gui: ShellGresourceFile     = /usr/share/gnome-shell/gnome-shell-theme.gresource
INFO:gdms.gui: UbuntuGdmGresourceFile = None
INFO:gdms.gui: Default Gresource File = /usr/share/gnome-shell/gnome-shell-theme.gresource.default
INFO:gdms.gui: pkexec version 123
```

tried again now shows this error:
`/run/user/1000/app/io.github.realmazharhussain.GdmSettings/run-elevated: 1: rm: not found`

output of whoami
```
echo $XDG_RUNTIME_DIR
echo $XDG_CACHE_HOME 
```

current username
/run/user/1000

**Reply by realmazharhussain** [reply]
This is not the AppImage. It seems like you installed the app from source as well. Running `gdm-settings` command would run that version instead. Hence the following lines in the output

```
INFO:gdms.gui: Application Version    = 4.alpha
INFO:gdms.gui: PackageType            = Unknown
```

Please! Uninstall all versions of the app and then install and run a single version.

**Reply by realmazharhussain** [reply]
Also, what is the output of `pkexec sh -c 'echo $PATH; which cp rm'`?

**Comment by farhan-beg** [comment]
uninstalled all installed from software app still same issue. 
output of `pkexec sh -c 'echo $PATH; which cp rm'`:
```
$PATH:$JAVA_HOME/bin
/usr/bin/sh: 1: which: not found
```
output of  `which cp rm` when ran as seperate command:
```
/usr/bin/cp
/usr/bin/rm
```
output of `pkexec sh -c 'echo $PATH'; which cp rm`:  
```
$PATH:$JAVA_HOME/bin
/usr/bin/cp
/usr/bin/rm
```

**Reply by realmazharhussain** [reply]
Is there any difference in the output if you run `pkexec bash -c 'echo $PATH; which cp rm'` (*sh is replaced with bash*) instead?

**Reply by farhan-beg** [reply]
no, no difference.

**Reply by realmazharhussain** [reply]
There is definitely something wrong with your polkit configuration/installation.

**Reply by farhan-beg** [reply]
what do u suggest i should do??

**Reply by farhan-beg** [reply]
i have sent u a msg on matrix if u want we can move this discussion there as u say this is not a issue of app but my system

**Reply by realmazharhussain** [reply]
First of all, do a reinstall of polkit with `sudo apt reinstall policykit-1`. Then remove every file from `/etc/polkit-1/rules.d/`. Restart and then try again.

**Reply by realmazharhussain** [reply]
> i have sent u a msg on matrix if u want we can move this discussion there as u say this is not a issue of app but my system

We can continue the discussion on GitHub. It's fine.

**Reply by farhan-beg** [reply]
did what u said still same issue no files in rules.d and reinstalled it

**Reply by realmazharhussain** [reply]
I don't have any more ideas. Maybe reach out on Kali Linux forums or bug trackers, etc.?

**Reply by farhan-beg** [reply]
sad, do u have any alternatives, which could work?

**Reply by realmazharhussain** [reply]
Not for polkit but for gdm-settings, there exists [gdm-tools](https://github.com/realmazharhussain/gdm-tools). It is deprecated but it does exist.

**Reply by farhan-beg** [reply]
It worked thnx sir
